### DISCUSSION TITLE
How to always push text when at the beginning of a line?

### DISCUSSION THREAD

**Discussion by yanshay** [abecodes/tabout.nvim#59] [Q&A] [answered]
I want that when the cursor is at the beginning of a line (so on the first non space character) to have the tab push the text like standard tab does. Currently on most lines it jumps to some other position in the line.
How can I achieve this?

**Comment by abecodes** [comment, marked as answer]
Hi @yanshay ,

```
    act_as_tab = true, -- shift content if tab out is not possible
    act_as_shift_tab = false, -- reverse shift content if tab out is not possible (if your keyboard/terminal supports <S-Tab>)
    default_tab = '<C-t>', -- shift default action (only at the beginning of a line, otherwise <TAB> is used)
    default_shift_tab = '<C-d>', -- reverse shift default action,
```

These settings are your friend.
Normally `act_as_tab` should already capture this behavior, but only, if it is truly the beginning of the line.
If it is after a tab etc, you need to hit <C(trl)-t> to get a tab forward and a <C(trl)-d> for backwards. Both are standard in (n)vim.

Happy holidays and stay healthy

**Reply by yanshay** [reply]
I was referring to the case when the cursor is not on the beginning of the line but rather on the first non-space character.
I understand this is not supported, so I found a way to add some lua code to identify that case before sending the tab to Tabout.

**Reply by abecodes** [reply]
Hi @yanshay,

it is supported:

> If it is after a tab etc, you need to hit `<C(trl)-t> `to get a tab forward and a `<C(trl)-d> `for backwards. Both are standard in (n)vim.

No need for extra code :)

**Reply by yanshay** [reply]
I'm aware of that this alternative key-press and have been trying to use it for quite some time now but that's not what I'm looking for. 
 
I just couldn't get used to having to remember that when I want `<Tab>` I need to press `<C-t>` if cursor is on first row char, so kept making the mistake of pressing Tab,  having to go back and then press `<C-t>`. I prefer the computer to automate what I need, not force me to adjust to how it works.

In my usage I've never encountered a case where `Tabout` functionality was helpful when the cursor is on the first character of the line, maybe just my coding style. So coding this behavior was easier than forcing myself to learn a workaround to how I really want neovim to behave.

I think it's a pretty typical need for some users and worth adding as a feature to the plugin (just like the option in case cursor is on the first row column), but since it's easily fixed, it's not a big deal.

**Reply by abecodes** [reply]
What I meant was, that <C(trl)-t> and <C(trl)-d> are the default/standard way of n(vim) of doing this.

I agree about the feature, and it works most of the times as expected, but depends on multi line tabout and treesitter nodes.
But I am happy to take PRs, so feel free to improve 😊
